src/client/Microsoft.Identity.Client/Cache/Items/MsalIdTokenCacheItem.cs

```diff
@@ -63,7 +66,12 @@ internal MsalIdTokenCacheItem()
                                     IsAdfs ? string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, "adfs") :
                                     string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
 
-        internal IdToken IdToken => IdToken.Parse(Secret);
```

Comment:
This is a great example of dangerous property use :) Callers of this property imply that the token is already parsed. It's not. A method would have been better here. 

In the end I chose to cache the parsed value.

Reply:
why not change to a method, then?

Reply:
(perhaps because it would require additional changes in other places of the code. we generally use the lazy pattern in a number of places. do you prefer methods for this?)

---
In reply to: [401278439](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1735#discussion_r401278439) [](ancestors = 401278439)